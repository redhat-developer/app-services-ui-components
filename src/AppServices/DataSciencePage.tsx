import {
  Bullseye,
  Button,
  ButtonVariant,
  Flex,
  Grid,
  GridItem,
  List,
  ListItem,
  Modal,
  Radio,
  Select,
  SelectOption,
  Spinner,
  Stack,
  StackItem,
  Text,
  TextContent,
  Title,
  TitleSizes,
} from "@patternfly/react-core";
import { useTranslation } from "react-i18next";
import { RhodsMlTechnology } from "../images";
import {
  MarketingPageHero,
  MarketingPageSection,
  MarketingPageVideoCard,
} from "./components";
import { ExternalLinkAltIcon } from "@patternfly/react-icons";
import { useState, useCallback } from "react";
import type { VoidFunctionComponent } from "react";
import { ExternalLink } from "../shared";

export type ClusterObject = {
  /**
   * Do not list all properties
   * Only define the ones that are used and required bu the components
   * THat will prevent TS from blocking the object and enforce the right properties
   */
  [key: string]: any;
  id: string;
  cluster_id: string;
  display_name: string;
  metrics?: { nodes?: { compute: number } }[];
};

export type ClustersResponse = {
  clusters: ClusterObject[];
  installableClusters: ClusterObject[];
};

export enum RHodsClusterAddonMode {
  Detecting,
  Install,
  Upgrade,
  Create,
}

export type DataSciencePageProps = {
  loadClusters: () => Promise<ClustersResponse>;
};

type HandleSelectCluster = (cluster: ClusterObject) => void;
type ClusterSelectProps = {
  selectedCluster?: ClusterObject;
  clusters: ClusterObject[];
  handleSelectCluster: HandleSelectCluster;
  isDisabled?: boolean;
};
enum UpgradeStrategy {
  Upgrade,
  Create,
}

const CREATE_CLUSTER_HREF = `${document.baseURI}openshift/create `;
const UPGRADE_CLUSTER_HREF = `${document.baseURI}openshift/details/s/{subscriptionID}#machinePools`;
const INSTALL_ADDON_HREF = `${document.baseURI}openshift/details/s/{subscriptionID}#addOns`;

const ClusterSelect: VoidFunctionComponent<ClusterSelectProps> = ({
  clusters,
  handleSelectCluster,
  selectedCluster,
  isDisabled,
}) => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <Select
      isDisabled={isDisabled}
      data-testid="select-RHODS-cluster"
      onSelect={() => setIsOpen(false)}
      selections={
        selectedCluster
          ? {
              toString: () => selectedCluster.display_name,
              compareTo: (selectOption) =>
                selectOption === selectedCluster.cluster_id,
            }
          : undefined
      }
      isOpen={isOpen}
      menuAppendTo="parent"
      variant="single"
      onToggle={(isOpen) => setIsOpen(isOpen)}
    >
      {clusters.map((cluster) => (
        <SelectOption
          onClick={() => {
            setIsOpen(false);
            handleSelectCluster(cluster);
          }}
          key={cluster.cluster_id}
          value={{
            toString: () => cluster.display_name,
            compareTo: (selectOption) => selectOption === cluster.cluster_id,
          }}
        />
      ))}
    </Select>
  );
};

const InstallClusterModalContent: VoidFunctionComponent<ClusterSelectProps> = (
  props
) => {
  const { t } = useTranslation("datascienceoverview");
  return (
    <Stack hasGutter>
      <StackItem>
        <TextContent className="pf-u-mb-sm">
          <Text component="p">{t("selectClusterLabel")}</Text>
        </TextContent>
        <ClusterSelect {...props} />
      </StackItem>
      <StackItem>
        <TextContent>
          <Text>{t("installClusterContent")}</Text>
        </TextContent>
      </StackItem>
    </Stack>
  );
};

type UpgradeClusterModalContentProps = ClusterSelectProps & {
  upgradeStrategy: UpgradeStrategy;
  handleStrategyChange: (strategy: UpgradeStrategy) => void;
};

const UpgradeClusterModalContent: VoidFunctionComponent<
  UpgradeClusterModalContentProps
> = ({ handleStrategyChange, upgradeStrategy, ...props }) => {
  const { t } = useTranslation("datascienceoverview");
  const upgradeBody = (
    <TextContent>
      <Text component="small">{t("upgradeClusterSelectDescription")}</Text>
      <ClusterSelect
        {...props}
        isDisabled={upgradeStrategy !== UpgradeStrategy.Upgrade}
      />
      <TextContent>
        <Text>{t("upgradeClusterPrerequisites")}</Text>
        <List>
          <ListItem>{t("upgradeClusterNodes")}</ListItem>
          <ListItem>{t("upgradeClusterCPUs")}</ListItem>
          <ListItem>{t("upgradeClusterMemory")}</ListItem>
        </List>
      </TextContent>
    </TextContent>
  );
  return (
    <Stack hasGutter>
      <StackItem>
        <Radio
          id="upgrade-cluster"
          name="upgrade-cluster"
          data-testid="radio-RHODS-action-upgrade"
          label={t("upgradeClusterSelectLabel")}
          onChange={() => handleStrategyChange(UpgradeStrategy.Upgrade)}
          isChecked={upgradeStrategy === UpgradeStrategy.Upgrade}
          body={upgradeBody}
        />
      </StackItem>
      <StackItem>
        <Radio
          id="create-cluster"
          name="create-cluster"
          data-testid="radio-RHODS-action-create"
          label={t("createClusterLabel")}
          onChange={() => handleStrategyChange(UpgradeStrategy.Create)}
          isChecked={upgradeStrategy === UpgradeStrategy.Create}
        />
      </StackItem>
    </Stack>
  );
};

const CreateClusterContent = () => {
  const { t } = useTranslation("datascienceoverview");
  return (
    <TextContent>
      <Text>{t("createClusterContent")}</Text>
    </TextContent>
  );
};

type ClusterModalContentProps = {
  selectedCluster?: ClusterObject;
  handleSelectCluster: HandleSelectCluster;
  mode: RHodsClusterAddonMode;
  clusters: ClusterObject[];
  upgradeStrategy: UpgradeStrategy;
  handleStrategyChange: (strategy: UpgradeStrategy) => void;
};

const ClusterModalContent: VoidFunctionComponent<ClusterModalContentProps> = ({
  mode,
  clusters,
  handleSelectCluster,
  selectedCluster,
  upgradeStrategy,
  handleStrategyChange,
}) => {
  if (mode === RHodsClusterAddonMode.Detecting) {
    return (
      <Bullseye>
        <Spinner size="xl" />
      </Bullseye>
    );
  }

  if (mode === RHodsClusterAddonMode.Install) {
    return (
      <InstallClusterModalContent
        selectedCluster={selectedCluster}
        handleSelectCluster={handleSelectCluster}
        clusters={clusters}
      />
    );
  }

  if (mode === RHodsClusterAddonMode.Upgrade) {
    return (
      <UpgradeClusterModalContent
        selectedCluster={selectedCluster}
        handleSelectCluster={handleSelectCluster}
        clusters={clusters}
        upgradeStrategy={upgradeStrategy}
        handleStrategyChange={handleStrategyChange}
      />
    );
  }

  if (mode === RHodsClusterAddonMode.Create) {
    return <CreateClusterContent />;
  }

  return null;
};

export const DataSciencePage: VoidFunctionComponent<DataSciencePageProps> = ({
  loadClusters,
}) => {
  const { t } = useTranslation("datascienceoverview");
  const [isModalOpen, setIsModalOpen] = useState(false);
  const [selectedCluster, setSelectedCluster] = useState<
    ClusterObject | undefined
  >();
  const [upgradeStrategy, setUpgradeStrategy] = useState<UpgradeStrategy>(
    UpgradeStrategy.Upgrade
  );
  const [{ mode, clusters }, setClustersState] = useState<{
    clusters: ClusterObject[];
    mode: RHodsClusterAddonMode;
  }>({ clusters: [], mode: RHodsClusterAddonMode.Detecting });

  const modalActions = {
    [RHodsClusterAddonMode.Detecting]: [],
    [RHodsClusterAddonMode.Install]: [
      <Button
        variant="primary"
        key="install"
        component="a"
        data-testid="install-RHODS-button"
        href={INSTALL_ADDON_HREF.replace(
          "{subscriptionID}",
          selectedCluster?.id || ""
        )}
      >
        {t("installClusterAction")}
      </Button>,
      <Button
        onClick={() => setIsModalOpen(false)}
        key="cancel"
        variant="secondary"
        data-testid="install-RHODS-cancel"
      >
        {t("cancelClusterAction")}
      </Button>,
    ],
    [RHodsClusterAddonMode.Upgrade]: [
      <Button
        variant="primary"
        key="install"
        component="a"
        data-testid="upgrade-RHODS-button"
        href={
          upgradeStrategy === UpgradeStrategy.Create
            ? CREATE_CLUSTER_HREF
            : UPGRADE_CLUSTER_HREF.replace(
                "{subscriptionID}",
                selectedCluster?.id || ""
              )
        }
      >
        {upgradeStrategy === UpgradeStrategy.Create
          ? t("createClusterAction")
          : t("upgradeAction")}
      </Button>,
      <Button
        onClick={() => setIsModalOpen(false)}
        key="cancel"
        variant="secondary"
        data-testid="install-RHODS-cancel"
      >
        {t("cancelClusterAction")}
      </Button>,
    ],
    [RHodsClusterAddonMode.Create]: [
      <Button
        component="a"
        href={CREATE_CLUSTER_HREF}
        variant="primary"
        key="install"
      >
        {t("createClusterAction")}
      </Button>,
      <Button
        onClick={() => setIsModalOpen(false)}
        key="cancel"
        variant="secondary"
        data-testid="install-RHODS-cancel"
      >
        {t("cancelClusterAction")}
      </Button>,
    ],
  };

  const clusterModalDescriptions = {
    [RHodsClusterAddonMode.Install]: t("installClusterDescription"),
    [RHodsClusterAddonMode.Upgrade]: t("upgradeClusterDescription"),
    [RHodsClusterAddonMode.Create]: t("createClusterDescription"),
  };

  const clusterModalTitles = {
    [RHodsClusterAddonMode.Detecting]: t("detectingClustersTitle"),
    [RHodsClusterAddonMode.Install]: t("installClusterTitle"),
    [RHodsClusterAddonMode.Upgrade]: t("installClusterTitle"),
    [RHodsClusterAddonMode.Create]: t("installClusterTitle"),
  };

  const getData = useCallback(async () => {
    const { clusters, installableClusters } = await loadClusters();
    if (installableClusters.length > 0) {
      setClustersState({
        mode: RHodsClusterAddonMode.Install,
        clusters: installableClusters,
      });
    } else {
      setClustersState({
        mode:
          clusters.length > 0
            ? RHodsClusterAddonMode.Upgrade
            : RHodsClusterAddonMode.Create,
        clusters,
      });
    }
  }, [loadClusters]);

  const handleInstallModalOpen = () => {
    // eslint-disable-next-line @typescript-eslint/no-floating-promises
    getData();
    setIsModalOpen(true);
  };

  const handleSelectCluster = (cluster: ClusterObject) => {
    setSelectedCluster(cluster);
  };

  return (
    <>
      <Modal
        isOpen={isModalOpen}
        onClose={() => setIsModalOpen(false)}
        variant="medium"
        data-testid="data-science-modal"
        actions={modalActions[mode]}
        title={clusterModalTitles[mode]}
        description={
          mode !== RHodsClusterAddonMode.Detecting ? (
            <Title headingLevel="h2" size="md">
              {clusterModalDescriptions[mode]}
            </Title>
          ) : undefined
        }
      >
        <ClusterModalContent
          upgradeStrategy={upgradeStrategy}
          handleSelectCluster={handleSelectCluster}
          selectedCluster={selectedCluster}
          mode={mode}
          clusters={clusters}
          handleStrategyChange={setUpgradeStrategy}
        />
      </Modal>
      <MarketingPageHero
        title={t("heroTitle")}
        tagLine={t("heroTagline")}
        description={t("heroDescription")}
        cta={
          <Flex>
            <Button
              data-testid="hero-buttonInstall"
              ouiaId="button-rhods-install"
              onClick={handleInstallModalOpen}
            >
              {t("installButton")}
            </Button>
            <ExternalLink
              testId="hero-buttonTry"
              ouiaId="button-rhods-try"
              href="https://developers.redhat.com/products/red-hat-openshift-data-science/getting-started?extIdCarryOver=true&sc_cid=701f2000001Css5AAC"
            >
              {t("heroTryItButton")}
            </ExternalLink>
          </Flex>
        }
        heroImage={RhodsMlTechnology}
        heroImageSize={478}
        heroImageCanRepeat={false}
        heroImagePositionY={-99}
      />
      <MarketingPageSection>
        <Grid hasGutter>
          <GridItem md={5}>
            <Stack hasGutter>
              <StackItem>
                <TextContent>
                  <Title
                    size={TitleSizes.xl}
                    headingLevel="h3"
                    className="pf-u-mb-lg"
                  >
                    {t("videoSectionTitle")}
                  </Title>
                  <Text className="pf-u-mr-md">
                    {t("videoSectionInThisVideo")}
                  </Text>
                </TextContent>
              </StackItem>
              <StackItem>
                <Button
                  data-testid="CTA-videoRHODSDemo"
                  variant={ButtonVariant.secondary}
                  component="a"
                  href="https://www.openshift.com/DataScienceVideoDemo"
                  target="_blank"
                >
                  {t("heroViewDemo")}{" "}
                  <ExternalLinkAltIcon className="pf-u-ml-sm" />
                </Button>
              </StackItem>
            </Stack>
          </GridItem>
          <GridItem md={7}>
            <MarketingPageVideoCard
              src={"https://www.youtube.com/embed/joK89xYeuUY"}
              title={t("videoSectionTitle")}
            />
          </GridItem>
        </Grid>
      </MarketingPageSection>
    </>
  );
};
