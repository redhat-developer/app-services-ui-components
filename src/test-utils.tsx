import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
