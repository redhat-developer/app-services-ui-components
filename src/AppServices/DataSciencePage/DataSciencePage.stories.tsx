import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { DataSciencePage } from "./DataSciencePage";

export default {
  component: DataSciencePage,
  args: {},
} as ComponentMeta<typeof DataSciencePage>;

const Template: ComponentStory<typeof DataSciencePage> = (args) => (
  <DataSciencePage {...args} />
);

export const Example = Template.bind({});
