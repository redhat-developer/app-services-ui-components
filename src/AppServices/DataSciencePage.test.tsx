import { expect } from "@storybook/jest";
import { render, waitForI18n, fireEvent, waitFor, screen } from "../test-utils";
import type { ClustersResponse } from "./DataSciencePage";
import { DataSciencePage } from "./DataSciencePage";

const installAddonResponse: ClustersResponse = {
  clusters: [],
  installableClusters: [
    {
      id: "installable-subs-1",
      cluster_id: "installable-cluster-1",
      display_name: "Installable cluster 1",
    },
    {
      id: "installable-subs-2",
      cluster_id: "installable-cluster-2",
      display_name: "Installable cluster 2",
    },
  ],
};

const upgradeAddonResponse: ClustersResponse = {
  clusters: [
    {
      id: "upgradeable-subs-1",
      cluster_id: "upgradeable-cluster-1",
      display_name: "Upgradeable cluster 1",
    },
    {
      id: "upgradeable-subs-2",
      cluster_id: "upgradeable-cluster-2",
      display_name: "Upgradeable cluster 2",
    },
  ],
  installableClusters: [],
};

const noClustersResponse: ClustersResponse = {
  clusters: [],
  installableClusters: [],
};

function loadInstallableClusters() {
  return Promise.resolve(installAddonResponse);
}

function loadUpgradeableClusters() {
  return Promise.resolve(upgradeAddonResponse);
}

function loadNoClusters() {
  return Promise.resolve(noClustersResponse);
}

describe("DataSciencePage", () => {
  it("renders", async () => {
    const comp = render(<DataSciencePage loadClusters={loadNoClusters} />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();

    const btnSubmit = comp.getByTestId("hero-buttonInstall");
    expect(btnSubmit.dataset.ouiaComponentId).toBe("button-rhods-install");
  });

  test("should open and close modal", async () => {
    const comp = render(
      <DataSciencePage loadClusters={loadInstallableClusters} />
    );
    await waitForI18n(comp);
    const installBtn = comp.getByTestId("hero-buttonInstall");

    // modal is hidden
    expect(screen.queryByTestId("data-science-modal")).not.toBeInTheDocument();

    // click on install button to open modal
    await waitFor(() => {
      fireEvent.click(installBtn);
    });
    expect(screen.queryByTestId("data-science-modal")).toBeInTheDocument();

    // clock cancel button
    await waitFor(() => {
      fireEvent.click(comp.getByTestId("install-RHODS-cancel"));
    });

    expect(screen.queryByTestId("data-science-modal")).not.toBeInTheDocument();
  });

  test("should open modal in install mode", async () => {
    const comp = render(
      <DataSciencePage loadClusters={loadInstallableClusters} />
    );
    await waitForI18n(comp);
    const installBtn = comp.getByTestId("hero-buttonInstall");

    // modal is hidden
    expect(screen.queryByTestId("data-science-modal")).not.toBeInTheDocument();

    // click on install button to open modal
    await waitFor(() => {
      fireEvent.click(installBtn);
    });
    expect(screen.queryByTestId("data-science-modal")).toBeInTheDocument();
    expect(
      screen.queryByText("Install OpenShift Data Science on your cluster")
    ).toBeInTheDocument();
  });

  test("should select cluster to which install RHODS addon", async () => {
    const comp = render(
      <DataSciencePage loadClusters={loadInstallableClusters} />
    );
    await waitForI18n(comp);
    const installBtn = comp.getByTestId("hero-buttonInstall");
    // click on install button to open modal
    await waitFor(() => {
      fireEvent.click(installBtn);
    });
    // open select menu
    fireEvent.click(
      screen.getByText(installAddonResponse.installableClusters[0].display_name)
    );
    // select second option
    fireEvent.click(
      screen.getByText(installAddonResponse.installableClusters[1].display_name)
    );
    const installButton = screen.getByTestId("install-RHODS-button");
    // check if the action has correct link to second cluster
    expect(installButton).toHaveAttribute(
      "href",
      `http://localhost/openshift/details/s/${installAddonResponse.installableClusters[1].id}#addOns`
    );
  });

  test("should select cluster to upgrade", async () => {
    const comp = render(
      <DataSciencePage loadClusters={loadUpgradeableClusters} />
    );
    await waitForI18n(comp);
    const installBtn = comp.getByTestId("hero-buttonInstall");
    // click on install button to open modal
    await waitFor(() => {
      fireEvent.click(installBtn);
    });
    // open select menu
    fireEvent.click(
      screen.getByText(upgradeAddonResponse.clusters[0].display_name)
    );
    // select second option
    fireEvent.click(
      screen.getByText(upgradeAddonResponse.clusters[1].display_name)
    );
    const upgradeButton = screen.getByTestId("upgrade-RHODS-button");
    // check if the action has correct link to second cluster
    expect(upgradeButton).toHaveAttribute(
      "href",
      `http://localhost/openshift/details/s/${upgradeAddonResponse.clusters[1].id}#machinePools`
    );
  });

  test("change action in upgreadable modal flow", async () => {
    const comp = render(
      <DataSciencePage loadClusters={loadUpgradeableClusters} />
    );
    await waitForI18n(comp);
    const installBtn = comp.getByTestId("hero-buttonInstall");
    // click on install button to open modal
    await waitFor(() => {
      fireEvent.click(installBtn);
    });

    const actionButton = screen.getByTestId("upgrade-RHODS-button");
    // check the action is in Upgrade mode
    expect(actionButton).toHaveTextContent("Upgrade");
    await waitFor(() => {
      fireEvent.click(screen.getByTestId("radio-RHODS-action-create"));
    });
    // check the action is in UpgradeCreate
    expect(actionButton).toHaveTextContent("Create");
    await waitFor(() => {
      fireEvent.click(screen.getByTestId("radio-RHODS-action-upgrade"));
    });
    // check the action is back in UpgradeCreate
    expect(actionButton).toHaveTextContent("Upgrade");
  });

  test("open modal in create mode", async () => {
    const comp = render(<DataSciencePage loadClusters={loadNoClusters} />);
    await waitForI18n(comp);
    const installBtn = comp.getByTestId("hero-buttonInstall");
    // click on install button to open modal
    await waitFor(() => {
      fireEvent.click(installBtn);
    });

    const modal = screen.queryByTestId("data-science-modal");
    expect(modal).toMatchSnapshot();
  });
});
