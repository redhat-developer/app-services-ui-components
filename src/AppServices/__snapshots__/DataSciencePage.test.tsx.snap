// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSciencePage open modal in create mode 1`] = `
<div
  class="pf-c-modal-box__body"
  data-testid="data-science-modal"
>
  <div
    class="pf-c-content"
  >
    <p
      class=""
      data-ouia-component-id="OUIA-Generated-Text-32"
      data-ouia-component-type="PF4/Text"
      data-ouia-safe="true"
      data-pf-content="true"
    >
      To create a new cluster, you'll be redirected to Cluster Manager. Don't worry, you'll be redirected back here afterwards so you can install OpenShift Data Science.
    </p>
  </div>
</div>
`;

exports[`DataSciencePage renders 1`] = `
<body>
  <div>
    <section
      class="pf-c-page__main-section pf-m-light appsrv-marketing--banner pf-u-background-color-100"
      style="--appsrv-marketing--banner--before--BackgroundImage: url(dummyimage); --appsrv-marketing--banner--before--BackgroundSize: 478px; --appsrv-marketing--banner--before--BackgroundRepeat: no-repeat; --appsrv-marketing--banner--before--BackgroundPositionY: -99px;"
    >
      <div
        class="pf-l-stack pf-m-gutter"
      >
        <h1
          class="pf-c-title pf-m-2xl"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
        >
          Get started with Red Hat OpenShift Data Science
        </h1>
        <p
          class="appsrv-marketing--banner__tagline pf-u-color-200"
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Add-on service for managed OpenShift
        </p>
        <p
          class=""
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Accelerate your data science. OpenShift Data Science is a managed cloud service for data scientists and developers of intelligent applications. Within a fully supported sandbox, it allows businesses to rapidly develop, train, and test artificial intelligence (AI) and machine learning (ML) models using Red Hat and partner technology.
        </p>
        <div
          class="pf-l-stack__item"
        >
          <div
            class="pf-l-flex"
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-primary"
              data-ouia-component-id="button-rhods-install"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="hero-buttonInstall"
              type="button"
            >
              Install it on a cluster
            </button>
            <a
              aria-disabled="false"
              class="pf-c-button pf-m-link pf-m-inline"
              data-ouia-component-id="button-rhods-try"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="hero-buttonTry"
              href="https://developers.redhat.com/products/red-hat-openshift-data-science/getting-started?extIdCarryOver=true&sc_cid=701f2000001Css5AAC"
              target="_blank"
            >
              Try it in the sandbox
              <span
                style="white-space: nowrap;"
              >
                 
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-limit-width appsrv-marketing--page-section--marketing"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item pf-m-5-col-on-md"
          >
            <div
              class="pf-l-stack pf-m-gutter"
            >
              <div
                class="pf-l-stack__item"
              >
                <div
                  class="pf-c-content"
                >
                  <h3
                    class="pf-c-title pf-m-xl pf-u-mb-lg"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Demo of Red Hat OpenShift Data Science
                  </h3>
                  <p
                    class="pf-u-mr-md"
                    data-ouia-component-id="OUIA-Generated-Text-3"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    In this video you’ll learn how to quickly build and deploy artificial intelligence and machine learning models using Red Hat and partner technology tools.
                  </p>
                </div>
              </div>
              <div
                class="pf-l-stack__item"
              >
                <a
                  aria-disabled="false"
                  class="pf-c-button pf-m-secondary"
                  data-ouia-component-id="OUIA-Generated-Button-secondary-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  data-testid="CTA-videoRHODSDemo"
                  href="https://www.openshift.com/DataScienceVideoDemo"
                  target="_blank"
                >
                  View the demo
                   
                  <svg
                    aria-hidden="true"
                    class="pf-u-ml-sm"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
          <div
            class="pf-l-grid__item pf-m-7-col-on-md"
          >
            <article
              class="pf-c-card appsrv-marketing--video"
              data-ouia-component-id="OUIA-Generated-Card-1"
              data-ouia-component-type="PF4/Card"
              data-ouia-safe="true"
              id=""
            >
              <div
                class="appsrv-marketing--video__wrapper"
              >
                <iframe
                  allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                  allowfullscreen=""
                  frameborder="0"
                  src="https://www.youtube.com/embed/joK89xYeuUY"
                  title="Demo of Red Hat OpenShift Data Science"
                />
              </div>
            </article>
          </div>
        </div>
      </div>
    </section>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
  <div />
</body>
`;
